### PR TITLE
Add generate_virtual_pop function and stat_summary function

### LINKED ISSUE #20: Write generate_virtual_pop function

### LINKED ISSUE #23: Write stat_summary function

### LINKED ISSUE #38: Add Raises section to docstring
Per Lise review

### PR DESCRIPTION
**Pull request by tguo9**
Please review and test it. I will update the tests function tonight

### CODE REVIEW

File: samplingsimulatorpy/generate_virtual_pop.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    distribution_func : func
+        The function that 
+    *para : int
+        The parameters the distribution_func is using
+
+    Returns
+    -------
+    pd.DataFrame
+        The virtual population as a dataframe
+
```
Comment (lisebraaten) [inline samplingsimulatorpy/generate_virtual_pop.py:22]: Add Raises section to docstring, see example below

```suggestion
    Raises
    -------
    ValueError
        size input is a positive integer
```

File: samplingsimulatorpy/generate_virtual_pop.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    try:
+        distribution_func(*para, size)
+    except TypeError:
+        print('Please enter correct number of parameters for the  distribution function')
+        raise
+
+    pop = pd.DataFrame(distribution_func(*para, size))
+
+    return pop
```
Comment (lisebraaten) [inline samplingsimulatorpy/generate_virtual_pop.py]:
```suggestion
    return pop
    
```

Comment (lisebraaten) [reply samplingsimulatorpy/generate_virtual_pop.py]: new line after function - no new line at end of file error

File: samplingsimulatorpy/stat_summary.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        The list of parameters
+    Returns
+    -------
+    pd.DataFrame
+        The summary stats as a dataframe
+    Examples
+    --------
+    >>> from samplingsimulatorpy import stat_summary
+    >>> stat_summary(pop, samples, parameter)
+    """
```
Comment (lisebraaten) [inline samplingsimulatorpy/stat_summary.py:37]: add spacing in between sections in docstring (as you did in your first function)
also add Raises section

File: samplingsimulatorpy/stat_summary.py [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+        try:
+            pop_para = eval(parameter[i])(population)[0]
+            samples_para = eval(parameter[i])(samples)[0]
+        except AttributeError:
+            print('Please enter a valid parameter(s) of interest')
+            raise AttributeError
+
+        new_dict[parameter[i]] = [pop_para, samples_para]
+
+    return(pd.DataFrame(new_dict).set_index('data'))
```
Comment (lisebraaten) [inline samplingsimulatorpy/stat_summary.py]: new line at end of file

```suggestion
    return(pd.DataFrame(new_dict).set_index('data'))
    
```

**Review by lisebraaten — CHANGES_REQUESTED** [review]

File: samplingsimulatorpy/generate_virtual_pop.py [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+    """
+    Create a virtual population
+
+    Parameters
+    ----------
+    size : int
+        The size of the virtual population 
+    distribution_func : func
+        The function that
```
Comment (lisebraaten) [inline samplingsimulatorpy/generate_virtual_pop.py]: finish this sentence

**Review by lisebraaten — CHANGES_REQUESTED** [review]

File: samplingsimulatorpy/generate_virtual_pop.py (hunk 1 above)
Comment (tguo9) [reply samplingsimulatorpy/generate_virtual_pop.py:22]: Nice catch! Addressed #38

File: samplingsimulatorpy/generate_virtual_pop.py (hunk 2 above)
Comment (tguo9) [reply samplingsimulatorpy/generate_virtual_pop.py]: I totally missed this one. Thanks Lise!

File: samplingsimulatorpy/stat_summary.py (hunk 3 above)
Comment (tguo9) [reply samplingsimulatorpy/stat_summary.py:37]: Got it! I will update it

File: samplingsimulatorpy/stat_summary.py (hunk 4 above)
Comment (tguo9) [reply samplingsimulatorpy/stat_summary.py]: Same. Thanks a lot for this one

File: samplingsimulatorpy/generate_virtual_pop.py (hunk 5 above)
Comment (tguo9) [reply samplingsimulatorpy/generate_virtual_pop.py]: Shoot... My mistake. Sorry about that

**Review by lisebraaten — APPROVED** [review]
thanks for making these changes Tao, looks good to me!

**Pull request merged**